dvc/repo/__init__.py

```diff
@@ -179,28 +179,34 @@ def collect(self, target, with_deps=False, recursive=False, graph=None):
         import networkx as nx
         from dvc.stage import Stage
 
-        G = graph or self.graph
-
         if not target:
-            return list(G)
+            return list(graph) if graph else self.stages
```

Comment:
```suggestion
            return list(graph or self.stages)
```

Reply:
No need to make a shallow copy off stages.